### PR TITLE
Command exit code

### PR DESCRIPTION
**Pull request by pcattori**
addresses #157

### CODE REVIEW

**Comment by brettcannon** [comment]
While I think about whether I want to accept this, one issue with the PR as it stands is it doesn't differentiate between a successful run that has blockers and an actual error as Python exits with `1` if there's an uncaught exception.

**Comment by pcattori** [comment]
For the Travis-CI (also applies to something like TeamCity) use-case that I am referencing, I need to have `caniusepython3` exit with a non-zero code so that the test does not pass when there are dependencies that are not compatible with python3. That way as PRs come along in a project, a change that breaks compatibility with python3 via dependencies will incur a visible test failure.

I admit I don't know too much about exit codes. Would it be better if this caused an exit code other than `0` or `1`?

Or maybe there could be a flag to enable this sort of behavior: eg. if there are blockers, standard behavior will still return exit code of `0`, but with the `--strict` flag, it will return exit code of `1`

**Comment by brettcannon** [comment]
I don't have an answer as I don't typically use exit codes to do anything accept signal a failure. 😄 This is why `caniusepython3.check()` exists, so you can write an actual test in your test suite that will fail if you have dependencies that have not converted yet.

**Comment by pcattori** [comment]
looks like this is how `unittest` handles things as well: 

> By default main calls sys.exit() with an exit code indicating success or failure of the tests run.
https://docs.python.org/3/library/unittest.html#unittest.main

**Comment by pcattori** [comment]
now using `3` as the exit code as recommended by http://tldp.org/LDP/abs/html/exitcodes.html

**Comment by brettcannon** [comment]
Please also update `__main__.main()` to make the return code consistently used for all CLI uses.

**Comment by pcattori** [comment]
@brettcannon updated `__main__.main()` as described in your comment.

**Comment by brettcannon** [comment]
I'm happy with the solution, but I just realized there's no test. Any chance you could add one using something like `subprocess` to verify you get `3` as a return value if there are blockers?

**Comment by pcattori** [comment]
@brettcannon : Tests implemented. I decided on mocking out `dependencies.blockers` to test `sys.exit(3)` behavior. Let me know what you think.

File: caniusepython3/test/test_cli.py [hunk 1]
```diff
@@ -189,6 +189,16 @@ def test_verbose_output(self):
         ciu_main.projects_from_cli(['-v', '-p', 'ipython'])
         self.assertTrue(logging.getLogger('ciu').isEnabledFor(logging.INFO))
 
+    @mock.patch('caniusepython3.dependencies.blockers', lambda projects: ['blocker'])
+    def test_nonzero_return_code(self):
+        with tempfile.NamedTemporaryFile('w') as file:
+            file.write(EXAMPLE_REQUIREMENTS)
+            file.flush()
+            args = ['--requirements', file.name]
```
Comment (brettcannon) [inline caniusepython3/test/test_cli.py]: If you use `--projects` then you can avoid the temporary file.

Comment (pcattori) [inline caniusepython3/test/test_cli.py]: done

**Comment by pcattori** [comment]
@brettcannon should I increment the minor version to indicate a new feature?

**Comment by brettcannon** [comment]
I update the number at release time so there's no need to bump it now in this PR.

**Comment by brettcannon** [comment]
Thanks for the all the work you put into the change!

**Comment by pcattori** [comment]
Just curious: how often do you do releases for `caniusepython3`? I'm eager to `pip install` the new version and try this new feature out on Travis 😄

**Comment by brettcannon** [comment]
I make releases when I have time and there's a new feature. I can try to cut a new release sometime this week.

**Pull request merged**